From 4245eb6de0dea491667cebc2d616d695e4b00fcd Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Wed, 21 Jul 2021 07:17:45 -0500
Subject: [PATCH 7/9] HSD #14014535834: firmware: stratix10-svc: stop kernel
 thread once service is completed

commit 9c74bb0578a26dd295e8963d675cfd88871a38cd from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

Stop the service layer's kernel thread to save the CPU resources, whenever
the service layer's client completes its service.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/firmware/stratix10-svc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/firmware/stratix10-svc.c b/drivers/firmware/stratix10-svc.c
index 799da8ec8d6c..a45cb3527588 100644
--- a/drivers/firmware/stratix10-svc.c
+++ b/drivers/firmware/stratix10-svc.c
@@ -1094,8 +1094,8 @@ EXPORT_SYMBOL_GPL(stratix10_svc_send);
  */
 void stratix10_svc_done(struct stratix10_svc_chan *chan)
 {
-	/* stop thread when thread is running AND only one active client */
-	if (chan->ctrl->task && chan->ctrl->num_active_client <= 1) {
+	/* stop thread when thread is running */
+	if (chan->ctrl->task) {
 		pr_debug("svc_smc_hvc_shm_thread is stopped\n");
 		kthread_stop(chan->ctrl->task);
 		chan->ctrl->task = NULL;
-- 
2.31.1

